GunShipAlpha/src/GameState.java

```diff
@@ -6,39 +6,58 @@
 
 class GameState {
 
-    private CopyOnWriteArrayList<Enemy> enemies = new CopyOnWriteArrayList<>();
+    private CopyOnWriteArrayList<Behemoth> behemoths = new CopyOnWriteArrayList<>();
     private ArrayList<Line> lines = new ArrayList<>();
     private CopyOnWriteArrayList<Projectile> projectiles = new CopyOnWriteArrayList<>();
+    private CopyOnWriteArrayList<Tank> tanks = new CopyOnWriteArrayList<>();
 
 
-
-    void handleEnemyMovement() {
-        Iterator<Enemy> iterator = enemies.iterator();
+    void handleBehemothMovement() {
+        Iterator<Behemoth> iterator = behemoths.iterator();
         while (iterator.hasNext()) {
-            Enemy enemy = iterator.next();
-            if (!isEnemyBlocked(enemy)) {
+            Behemoth enemy = iterator.next();
+            if (!isBehemothBlocked(enemy)) {
                 enemy.moveTowardTarget();
                 removeEnemy();
             }
         }
     }
+    void handleTankMovement() {
+        Iterator<Tank> iterator = tanks.iterator();
+        while (iterator.hasNext()) {
+            Tank tank = iterator.next();
+            if (!isTankBlocked(tank)) {
+                tank.moveTowardTarget();
+                removeEnemy();
+            }
+        }
+    }
 
 
-    boolean isEnemyBlocked(Enemy enemy) {
+    boolean isBehemothBlocked(Behemoth enemy) {
```

Comment:
removing the tank/behemoth duplication will also allow you to remove this as well. I want you to see this kind of stuff and recognize it as wrong, like scattering tacks on your bed. duplication is bad.

Reply:
fixed